torchgeo/datasets/eurocrops.py

```diff
@@ -0,0 +1,285 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""EuroCrops dataset."""
+
+import csv
+import os
+from collections.abc import Iterable
+from typing import Any, Callable, Optional, Union
+
+import matplotlib.pyplot as plt
+import torch
+from matplotlib.figure import Figure
+from rasterio.crs import CRS
+
+from .geo import VectorDataset
+from .utils import (
+    DatasetNotFoundError,
+    check_integrity,
+    download_and_extract_archive,
+    download_url,
+)
+
+
+def split_hcat_code(hcat_code: str, hcat_level: int) -> tuple[str, str]:
+    """Splits the specified HCAT code into its prefix and suffix.
+
+    The code is split at the given hcat_level.
+    For example, a code 3301010100 at Level 4 has prefix "330101" and suffix "0100".
+
+    Args:
+        hcat_code: a class code like 3301010100.
+        hcat_level: the HCAT level (3, 4, 5, or 6).
+
+    Returns:
+        A tuple containing, respectively, the HCAT prefix and suffix. At Level 3, the
+        prefix is the first 4 digits and suffix is the second 6 digits, while at Level
+        6, the prefix is the full code and the suffix is empty.
+    """
+    assert len(hcat_code) == 10
+    if hcat_level == 3:
+        return hcat_code[0:4], hcat_code[4:10]
+    elif hcat_level == 4:
+        return hcat_code[0:6], hcat_code[6:10]
+    elif hcat_level == 5:
+        return hcat_code[0:8], hcat_code[8:10]
+    elif hcat_level == 6:
+        return hcat_code, ""
+    raise ValueError(f"Invalid HCAT level {hcat_level}.")
+
+
+class EuroCrops(VectorDataset):
+    """EuroCrops Dataset (Version 9).
+
+    The `EuroCrops <https://www.eurocrops.tum.de/index.html>`__ dataset combines "all
+    publicly available self-declared crop reporting datasets from countries of the
+    European Union" into a unified format. The dataset is released under CC BY 4.0 Deed.
+
+    The dataset consists of shapefiles containing a total of X polygons. Each polygon
```

Comment:
I assume X is a placeholder you plan on filling in

Reply:
Ah planned to fill it in before committing, fixed now!